_posts/2024-12-19-cosmicqc.md

```diff
@@ -0,0 +1,285 @@
+When working with single-cell data, we do not want these technical artifacts impacting the interpretation of our results.
+Thus, was born this simple idea in my head.
+
+Why not just filter out single cells based on the extracted morphology features?
+
+You wouldn't need to train any model or perform any other pipeline.
+You already have the data in your lap to use, but it is how you use that data is important.
+
+In comes, [coSMicQC](https://github.com/WayScience/coSMicQC), or Single cell Morphology Quality Control!
+
+This new method works by creating conditions, represented by dictionaries.
+A condition is a feature or group of features that will detect a specific type of technical artifact.
+For example, nuclei area and intensity can work together as a condition to detect large nuclei with high intensity which represents segmentation of clustered/overlapping nuclei.
+You condition will also need to include the standard deviation from the mean in which you expect these outliers to be located.
+If you want to detect cells above a mean, you use a positive value, and vice versa for cells below the mean.
+
+{%
+  include figure.html
+  image="images/blog/cosmicqc/area_outlier_example.png"
+  width="100%"
+%}
+> Example of how outliers are detected based on nuclei area in coSMicQC.
+
+We then apply the function `find_outliers` to the simple conditions independently, yielding a dataframe with only the identified outliers, including metadata and the features used to detect the outliers.
+A summary is also outputted, which includes the total number of outliers detected, the percent of the total cells in the data detected as outliers, and the range of the values of the outliers for the features used in the condition.
+
+A highlight of coSMicQC is that you are able to visualize single-cell segmentations within the notebook if you provide the original images and exported outlines from CellProfiler, through the use of a dataframe format called [CytoDataFrame](https://github.com/WayScience/CytoDataFrame).
+This makes it more streamlined to assess if your conditions are working without having to go outside of the notebook via your file manager, Fiji, or CellProfiler to see what the segmented cells look like.
+This feature is still being developed as we find more datasets with varying conditions for image file structure.
+
+Once you have optimized the conditions, you can then use the function `label_outliers` which will output the original dataframe with columns for each condition and if a single-cell passed or failed each condition.
+You can then choose to either save the original dataframe with the new columns and keep outliers or remove single cells that were detected as outliers.
+You can also not use the coSMicQC functionality and take the indices from the `find_outliers` outputs and remove single cells that way.
+We have elected to call the outputs that come from coSMicQC as `cleaned` profiles.
```

Comment:
I recommend you shift the structure of this slightly.

1. Describe, in general what coSMicQC does.
2. Describe, in detail, the different functions of coSMicQC to achieve this general function

Reply:
I will try to make a stab at this.